plugins/modules/groups.py

```diff
@@ -64,29 +89,93 @@ def main():
 
     create_group_data = {"name": module.params["name"], "type": "static"}
 
-    def find_group(group_data):
-        if group_data["data"] is None:
+    def find_group(group_data, name: str = module.params['name']):
+        if group_data['data'] == None:
```

Comment:
idiomatically this should be `is None` for the conditional, many linter tools will yell at you about this conditional